lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2300,21 +2300,36 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
 
   case OpFMod: {
     // translate OpFMod(a, b) to:
-    //   r = frem(a, b)
-    //   c = copysign(r, b)
-    //   needs_fixing = islessgreater(r, c)
-    //   result = needs_fixing ? r + b : c
+    //   r = fmod(a, b)
+    //   result = copysign(r, b)
+    // this is how the CTS expects it to be implemented
     IRBuilder<> Builder(BB);
     SPIRVFMod *FMod = static_cast<SPIRVFMod *>(BV);
-    auto Dividend = transValue(FMod->getOperand(0), F, BB);
     auto Divisor = transValue(FMod->getOperand(1), F, BB);
-    auto FRem = Builder.CreateFRem(Dividend, Divisor, "frem.res");
-    auto CopySign = Builder.CreateBinaryIntrinsic(
-        llvm::Intrinsic::copysign, FRem, Divisor, nullptr, "copysign.res");
-    auto FAdd = Builder.CreateFAdd(FRem, Divisor, "fadd.res");
-    auto Cmp = Builder.CreateFCmpONE(FRem, CopySign, "cmp.res");
-    auto Select = Builder.CreateSelect(Cmp, FAdd, CopySign);
-    return mapValue(BV, Select);
+    auto *FModBuiltinCall = transBuiltinFromInst("fmod", FMod, BB);
+    auto *CopySign = Builder.CreateBinaryIntrinsic(llvm::Intrinsic::copysign,
+                                                   FModBuiltinCall, Divisor,
+                                                   nullptr, "copysign.res");
+    return mapValue(BV, CopySign);
+  }
+
+  case OpFRem: {
+    // translate OpFRem(a, b) to:
+    //   result = fmod(a, b)
```

Comment:
This doesn't seem right, because result of [`OpFRem`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpFRem) should not be always the same as [`OpFMod`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpFMod): they both compute [Reminder](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Remainder), but `OpFRem` uses the sign of `Operand 1` and `OpFMod` uses the sign of `Operand 2`.

The comment says that results of those two are equal, which is clearly may not be the case per SPIR-V spec.